lib/msal-node/src/client/ClientCredentialClient.ts

```diff
@@ -89,6 +89,21 @@ export class ClientCredentialClient extends BaseClient {
             this.serverTelemetryManager?.setCacheOutcome(
                 CacheOutcome.CACHED_ACCESS_TOKEN_EXPIRED
             );
+
+            return null;
+        // must refresh due to the refresh_in value
+        } else if (
```

Comment:
I don't think this is the correct implementation, because it doesn't help with availability. If ESTS goes down during the token refresh phase, it kills the app. Please read the spec, particularly the acceptance tests.

The scenario that `refresh_in` is trying to protect against is the following: 

1. app gets a token from ESTS that expires in 60 min and it uses the token over and over (from the cache)
2. cached token expires and request goes to ESTS but **ESTS is down** 
3. app is no longer able to function, because it doesn't have a token to call some downstream api

With the introduction of refresh_in, the way it's supposed to work is: 

1. app gets a token from ESTS that expires in 60 min and with refresh_in=30min, at 12:00
2. between 12:00 and 12:30, the app can get the token out of the cache, just like before
3. at 12:40 app makes a request for the token. MSAL gives the app the token from the cache, but at the same time it sends a request to ESTS for a new token. Now the token is fresh.

With this, a fresh token is always in the cache, as long as the app uses it frequently. If ESTS goes down at step 3 above, MSAL should just ignore that and return the token from the cache.

Definition of "ESTS is down" is that it responds with 429 or 5xx errors (maybe other HTTP errors), but not with other 4xx.